base/applications/taskmgr/taskmgr.c

```diff
@@ -482,14 +482,23 @@ TaskManagerWndProc(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
         TrayIcon_UpdateIcon();
         break;
 
+    case WM_INITMENUPOPUP:
+        /* Do not disable the status bar if we opened the system menu */
+        if (!HIWORD(lParam))
+            TaskManager_DisableStatusBar(hDlg);
+        else
+            TaskManager_EnableStatusBar(hDlg);
+        break;
     case WM_ENTERMENULOOP:
-        TaskManager_OnEnterMenuLoop(hDlg);
+        bInMenuLoop = TRUE;
         break;
     case WM_EXITMENULOOP:
-        TaskManager_OnExitMenuLoop(hDlg);
+        TaskManager_EnableStatusBar(hDlg);
+        bInMenuLoop = FALSE;
```

Comment:
I think `bInMenuLoop = FALSE;` should be done before calling `TaskManager_EnableStatusBar()`, because the latter invokes `RefreshPerformancePage()`, which reads the value of that variable and behaves differently depending on it.